portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java

```diff
@@ -22,45 +8,68 @@
 import java.util.List;
 import com.google.gson.Gson;
 import com.google.sps.data.CommentsDatastore;
+import com.google.sps.data.Comment;
+import java.util.logging.Logger;
+import java.util.logging.Level;
 
 /** Servlet that handles comments data. */
 @WebServlet("/comments")
 public final class CommentsServlet extends HttpServlet {
+    private static final Logger log = Logger.getLogger(CommentsServlet.class.getName());
 
   /** Class that creates an object to hold the comments
   and limit on the number of comments displayed. */
   public final class CommentsApiResponse {
-    private List<CommentsDatastore.Comment> comments;
+    private final List<Comment> comments;
 
-    public CommentsApiResponse(List<CommentsDatastore.Comment> comments) {
+    public CommentsApiResponse(List<Comment> comments) {
       this.comments = comments;
     }
 
     /** Returns list of comments. */
-    public List<CommentsDatastore.Comment> getComments() {
+    public List<Comment> getComments() {
       return this.comments;
     }
   }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Integer commentsLimit = Integer.parseInt(request.getParameter("commentsLimit"));
+    String commentsLimitStr = (String) request.getParameter(CommentsDatastore.COMMENTS_LIMIT_PARAMETER);
+    if (commentsLimitStr == null || commentsLimitStr.isEmpty()) {
+      throw new IllegalArgumentException("comments limit should not be empty");
```

Comment:
you don't want to throw here (you are not in your try/catch), you want to return 400